flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategyTest.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.flink.runtime.state.StateHandleID;
+import org.apache.flink.runtime.state.StreamStateHandle;
+import org.apache.flink.runtime.state.TestLocalRecoveryConfig;
+import org.apache.flink.runtime.state.filesystem.FsCheckpointStreamFactory;
+import org.apache.flink.util.ResourceGuard;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import static org.apache.flink.core.fs.Path.fromLocalFile;
+import static org.apache.flink.core.fs.local.LocalFileSystem.getSharedInstance;
+import static org.powermock.api.mockito.PowerMockito.spy;
+
+/**
+ * Test for verifying the changed logic about notifyCheckpointComplete over
+ * RocksIncrementalSnapshotStrategy is incremental.
+ */
+public class RocksIncrementalSnapshotStrategyTest {
+
+    @Rule public final TemporaryFolder tmp = new TemporaryFolder();
+
+    @Test
+    public void testCheckpointIsIncremental() throws Exception {
+        CloseableRegistry closeableRegistry = new CloseableRegistry();
+
+        // prepare components
+        Tuple2<RocksIncrementalSnapshotStrategy, FsCheckpointStreamFactory> prepareData =
+                prepareCreateSnapshotStrategy(closeableRegistry);
+        RocksIncrementalSnapshotStrategy checkpointSnapshotStrategy = prepareData.f0;
+        FsCheckpointStreamFactory checkpointStreamFactory = prepareData.f1;
+
+        // make and notify checkpoint with id 1
+        snapshot(1L, checkpointSnapshotStrategy, checkpointStreamFactory, closeableRegistry);
+        checkpointSnapshotStrategy.notifyCheckpointComplete(1L);
+
+        // notify savepoint with id 2
+        checkpointSnapshotStrategy.notifyCheckpointComplete(2L);
+
+        // make checkpoint with id 3
+        IncrementalRemoteKeyedStateHandle incrementalRemoteKeyedStateHandle3 =
+                snapshot(
+                        3L, checkpointSnapshotStrategy, checkpointStreamFactory, closeableRegistry);
+
+        // If 3rd checkpoint's placeholderStateHandleCount > 0,it means 3rd checkpoint is
+        // incremental.
+        Map<StateHandleID, StreamStateHandle> sharedState3 =
+                incrementalRemoteKeyedStateHandle3.getSharedState();
+        long placeholderStateHandleCount =
+                sharedState3.entrySet().stream()
+                        .filter(e -> e.getValue() instanceof PlaceholderStreamStateHandle)
+                        .count();
+
+        Assert.assertTrue(placeholderStateHandleCount > 0);
+
+        checkpointSnapshotStrategy.close();
+        closeableRegistry.close();
```

Comment:
These closeables could be closed in `finally` statements to ensure no resource leak.